crates/bevy_ui/src/ui_node.rs

```diff
@@ -263,6 +265,7 @@ impl Default for Style {
             max_size: Size::AUTO,
             aspect_ratio: Default::default(),
             overflow: Default::default(),
+            gap: Size::AUTO,
```

Comment:
what does this mean? There is no support for `auto` value for `gap`, it should be a length, `normal`, `inherit` or `initial`

Reply:
Yeah, the default here should either be `Size::NONE` (which corresponds to not set) or `0.0` in both dimensions. The CSS spec has `normal`, but `normal` always corresponds to `0.0` except in multi-text-column layouts, so I didn't both with it. Possibly `gap` should be using a new `LengthPercentage` type that doesn't have `Auto` or `Undefined` variants rather than the existing `Dimension` type, but that isn't how it is currently implemented.

`Auto` and `Undefined` are both just treated as `0.0` when computing layout.

Reply:
> Possibly gap should be using a new LengthPercentage type that doesn't have Auto or Undefined variants rather than the existing Dimension type, but that isn't how it is currently implemented.

I would prefer to see that fixed in Taffy before being used in Bevy, or at least documented. As we refer user to CSS docs for how to work with `bevy_ui`, it doesn't feel good to add things that don't work the same way

Reply:
We could introduce the new type in Bevy, and map it to Taffy's Size to not block this update

Reply:
That seems like a good idea. It'll be a bit awkward to change in taffy immediately given that this a breaking change and we've only just put out a release. I would suggest something like:

```rust
enum LengthPercentage {
    /// Abstract absolute units
    Points(f32),
    /// The dimension is stored in percentage relative to the parent item.
    Percent(f32),
}
```

Reply:
Bevy uses `Px` to map to Taffy `Points`, but otherwise yes 👍

Reply:
> `Auto` and `Undefined` are both just treated as 0.0 when computing layout.

There are probably other values where `Auto` possible and nonsensical, right? `border` comes to mind.

I'd personally prefer to document this behavior and follow Taffy's lead on any new types introduced to clear this sort of thing up over introducing another new thing just for this property that will change in Taffy later anyhow.

Reply:
I've opened https://github.com/DioxusLabs/taffy/pull/269 to address this issue for all style properties.